dlt/destinations/job_client_impl.py

```diff
@@ -523,36 +520,137 @@ def _from_db_type(
-                f" ORDER BY {c_inserted_at} DESC;"
-            )
-            return self._row_to_schema_info(query)
-        else:
-            query = (
-                f"SELECT {self.version_table_schema_columns} FROM {name} WHERE {c_schema_name} = %s"
-                f" ORDER BY {c_inserted_at} DESC;"
+        table_name = self.schema.version_table_name
+        dataset_name = self.sql_client.dataset_name
+        catalog_name = self.sql_client.catalog_name(escape=False)
+
+        if self.config.destination_type == "clickhouse":
+            dataset_table_separator = self.config.dataset_table_separator  # type: ignore[attr-defined]
+            table_name = dataset_table_separator.join([dataset_name, table_name])
+            dataset_name = catalog_name
+            catalog_name = None
+
+        c_schema_name, c_inserted_at = [
+            self.schema.naming.normalize_path(p) for p in ["schema_name", "inserted_at"]
+        ]
+
+        table_expr = sqlglot.exp.Table(
+            this=sqlglot.exp.to_identifier(table_name, quoted=True),
+            db=sqlglot.exp.to_identifier(dataset_name, quoted=True),
+            catalog=sqlglot.exp.to_identifier(catalog_name, quoted=True),
+        )
+
+        select_expr = sqlglot.exp.Select(
+            expressions=[
+                sqlglot.exp.Column(this=sqlglot.exp.to_identifier(col, quoted=True))
+                for col in self.version_table_schema_cols
+            ]
+        )
+
+        order_expr = sqlglot.exp.Ordered(
+            this=sqlglot.exp.Column(this=sqlglot.exp.to_identifier(c_inserted_at, quoted=True)),
+            desc=True,
+        )
+
+        where_condition = None
+        if schema_name:
+            where_condition = sqlglot.exp.EQ(
+                this=sqlglot.exp.Column(this=c_schema_name),
+                expression=sqlglot.exp.Literal.string(schema_name),
             )
-            return self._row_to_schema_info(query, schema_name)
+
+        select_expr = select_expr.from_(table_expr).where(where_condition).order_by(order_expr)
+
+        dialect = self.sql_client.capabilities.sqlglot_dialect
+
+        query = select_expr.sql(dialect=dialect)
+
+        return self._row_to_schema_info(query)
 
     def get_stored_state(self, pipeline_name: str) -> StateInfo:
-        state_table = self.sql_client.make_qualified_table_name(self.schema.state_table_name)
-        loads_table = self.sql_client.make_qualified_table_name(self.schema.loads_table_name)
-        c_load_id, c_dlt_load_id, c_pipeline_name, c_status = self._norm_and_escape_columns(
-            "load_id", C_DLT_LOAD_ID, "pipeline_name", "status"
+        state_table_name = self.schema.state_table_name
+        loads_table_name = self.schema.loads_table_name
+        dataset_name = self.sql_client.dataset_name
+        catalog_name = self.sql_client.catalog_name(escape=False)
+
+        if self.config.destination_type == "clickhouse":
```

Comment:
Just on a general note, the destination specific implementations should never leak into the base class. The right way to do it would be to have the correct implementation in the subclass. But with the changes I propose below, this problem should be gone in any case.